control-plane/connect-inject/controllers/pod/pod_controller_test.go

```diff
@@ -1446,6 +2664,34 @@ func createProxyConfiguration() *pbmesh.ProxyConfiguration {
 	}
 }
 
+// createCriticalHealthStatus creates a failing HealthStatus that matches the pod from createPod.
+func createUpstreams() *pbmesh.Upstreams {
+	return &pbmesh.Upstreams{
+		Workloads: &pbcatalog.WorkloadSelector{
+			Names: []string{"foo"},
+		},
+		Upstreams: []*pbmesh.Upstream{
+			{
+				DestinationRef: &pbresource.Reference{
+					Tenancy: &pbresource.Tenancy{
+						Partition: "",
+						Namespace: "",
+						PeerName:  "",
```

Comment:
These should be explicit all the time. Part of this is historical that Consul V2 didn't default them, but if you set one of them, the others won't correctly inherit the default.